internal/coreinternal/processor/filterconfig/config.go

```diff
@@ -144,6 +144,19 @@ func (mp *MatchProperties) ValidateForLogs() error {
 	return nil
 }
 
+// ValidateForMetrics validates properties for metrics.
+func (mp *MatchProperties) ValidateForMetrics() error {
+	if len(mp.SpanNames) > 0 || len(mp.Services) > 0 || len(mp.LogNames) > 0 {
+		return errors.New("none of services, span_names nor log_names should be specified for metrics records")
+	}
+
+	if len(mp.Attributes) == 0 && len(mp.Libraries) == 0 && len(mp.Resources) == 0 && len(mp.MetricNames) == 0 {
+		return errors.New(`at least one of "attributes", "libraries", "resources" or "metric_names" field must be specified`)
```

Comment:
We would sync [wording of `README.md` for metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/attributesprocessor/README.md?plain=1#L158)